File: HelpfulMaths.java
import java.util.Scanner;
import java.util.ArrayList;
import java.util.Collections;

public class HelpfulMaths {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        String s = scanner.nextLine();
        ArrayList<Integer> numbers = new ArrayList<>();
        
        for (int i = 0; i < s.length(); i += 2) {
            numbers.add(s.charAt(i) - '0');
        }
        
        Collections.sort(numbers);
        
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < numbers.size(); i++) {
            result.append(numbers.get(i));
            if (i < numbers.size() - 1) {
                result.append("+");
            }
        }
        
        System.out.println(result);
        scanner.close();
    }
}

File: HelpfulMaths.cpp
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

int main() {
    string s;
    cin >> s;
    
    int numbers[s.length() / 2 + 1];
    int index = 0;
    
    for (int i = 0; i < s.length(); i += 2) {
        numbers[index++] = s[i] - '0';
    }
    
    sort(numbers, numbers + index);
    
    for (int i = 0; i < index; i++) {
        cout << numbers[i];
        if (i < index - 1) {
            cout << "+";
        }
    }
    
    return 0;
}